test/string_scanner_test.dart

```diff
@@ -393,6 +436,57 @@ void main() {
     });
   });
 
+  group('before a surrogate pair', () {
+    final codePoint = '👭'.runes.first;
```

Comment:
Consider writing out the string as an escape. A *reader* may want to know what code units the string contains.
I'd probably go with.
```dart
  final codePoint = '\ud83d\udc6d"; // "👭"
```